### PR TITLE
David/eslint simple import sort

### PR DESCRIPTION
**Pull request by haveanicedavid**
## Description

EDIT: Will not merge until: 
- [x] `.vscode.settings.suggested` is clarified 
- [ ] huksy is set up for pre-commit hook
- [x] current outstanding PRs are merged first

EDIT 2: Setting up husky wasn't as straightforward as I hoped to be. Could be worth a follow-up task, but in the immediate I'm not sure it will be an issue as it's pretty easy to to configure IDEs to format on save, and the codebase has been formatted with the new rules

Was curious how my personal eslint rules regarding imports / export sorting would apply to our codebase. Only took a few minutes to set up + some small tweaks... seeing if the build will pass.

Basically, it'll automatically rearrange imports (assuming your IDE is configured to format through `eslint --fix` into groups separated by a space (similar to our current pattern). I also globally updated an eslint react hook rule, as we had some dated version that wasn't allowing optional chaining in react effect deps

Obviously, this is a big PR as it's run on the whole codebase. The config changes show in the f[irst commit](https://github.com/regen-network/regen-web/pull/1075/commits/ea2a67241779649f7819aeabb2ddf21e0cb3b4b3), and you can check any file to see an example of how things are rearranged. 

The advantage of merging this would be:
1. We'd never have to manually organize imports again
2. conflicts with imports wouldn't be as common

IMO it'd be pretty safe to merge, as it only really touches imports, but did it mainly to discuss. It's minor, but manually arranging imports is pretty annoying

---

### Author Checklist

_All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues._

I have...

- [ ] provided a link to the relevant issue or specification
- [ ] provided instructions on how to test
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed
- [ ] once the PR is closed, set up backport PRs for `redwood` and `hambach` (see below)

#### Setting up backport PRs

After merging your PR to `master`, set up backports by doing the following:

1. If your branch does not have merge commits, add the following comment to
   your PR, `@Mergifyio backport redwood hambach`.

4. If your branch does have merge commits:

    a. Pull latest `master`, `hambach` and `redwood` branches

    b. Create new branches for backports and merge `master` (replace `<PR#>` with your PR #)
    ```
    git checkout -b hambach-backport-<PR#> hambach
    git merge master
    git push origin hambach-backport-<PR#>
    git checkout -b redwood-backport-<PR#> redwood
    git merge master
    git push origin redwood-backport-<PR#>`
    ```

    c. Open new PRs in regen-web targeting `hambach` and `redwood`, respectively.
  

### How to test

1.

### Reviewers Checklist

_All items are required. Please add a note if the item is not applicable and please **add
your handle next to the items reviewed if you only reviewed selected items**._

I have...

- [ ] confirmed all author checklist items have been addressed
- [ ] reviewed code correctness and readability
- [ ] verified React components follow DRY principles
- [ ] reviewed documentation is accurate
- [ ] reviewed tests
- [ ] manually tested (if applicable)

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -3,15 +3,44 @@ module.exports = {
...
     ],
   },
+  overrides: [
+    {
+      files: ['*.tsx', '*.ts'],
+      rules: {
+        'simple-import-sort/imports': [
+          'error',
+          {
+            groups: [
```
Comment (haveanicedavid) [inline .eslintrc.js:30]: how groupings are set (each group is separated by a space)

**Review by flagrede — APPROVED** [review]
Looks good, looking forward to having our imports all sorted!

**Review by mhagel — APPROVED** [review]

**Comment by haveanicedavid** [comment]
@Mergifyio backport redwood hambach v4

**Comment by haveanicedavid** [comment]
@mergifyio backport v4

**Pull request merged**